src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingMiner.java

```diff
@@ -186,16 +187,28 @@ public CompoundNBT serializeNBT()
         final CompoundNBT compound = super.serializeNBT();
         compound.putBoolean(TAG_CLEARED, clearedShaft);
 
-        if (cobbleLocation != null)
-        {
-            BlockPosUtil.write(compound, TAG_CLOCATION, cobbleLocation);
-        }
+        writeOptionalBlockPos(compound, TAG_CLOCATION, cobbleLocation);
+        writeOptionalBlockPos(compound, TAG_LLOCATION, ladderLocation);
+
+        return compound;
+    }
 
-        if (ladderLocation != null)
+    @Nullable
+    private static BlockPos readOptionalBlockPos(@NotNull final CompoundNBT compound,
```

Comment:
Those should maybe go to BlockPosUtil.

Reply:
Maybe, although the implementation of `readOptionalBlockPos` is deliberately "incorrect" for general purposes (it treats ZERO == null, in order to fix already-broken miners).  Still, there's other places (parent-child buildings, for example) that treat ZERO == null, so I guess it'd be in good company.